src/powerpc/ffi.c

```diff
@@ -183,6 +183,12 @@ ffi_tramp_arch (size_t *tramp_size, size_t *map_size)
   extern void *trampoline_code_table;
   *tramp_size = PPC_TRAMP_SIZE;
   *map_size = PPC_TRAMP_MAP_SIZE;
+#ifdef _CALL_AIX
```

Comment:
Clang doesn't define this FWIW. I would suggest

```suggestion
#if _CALL_ELF != 2
```

which is what's done in other parts of the PowerPC code.

Reply:
But that won't work either, because this file is used for all powerpc ABI and _CALL_ELF != 2 is true for 32-bit ppc and we don't want that.  The other uses of `#if _CALL_ELF != 2` are in 64-bit only code, so doesn't affect 32-bit compiles.

Does `#if defined (_CALL_AIX) || __CALL_ELF == 1` work with clang?

Reply:
Ah, good point. Perhaps just

```suggestion
#if _CALL_ELF == 1
```

then? I really don't think you need `_CALL_AIX` at all; GCC always defines it together with `_CALL_ELF=1`.

Reply:
It does now, but that hasn't always been the case.  Before we added `powerpc64le-linux`, we didn't emit `_CALL_ELF` at all, so if we built libffi with an old gcc on `powerpc64-linux`, then we'd fall into the `#else` branch of the code and we'd hit the same problem again.  I'd prefer using the version that checks for `_CALL_AIX` too in case we're building with an older gcc.

..but thank you for the info about clang not supporting `_CALL_AIX`!

Reply:
Ah, I wasn't aware of that bit of history, good to know! In that case, I agree with your `#if defined (_CALL_AIX) || _CALL_ELF == 1` suggestion.